db/fetcher/encoded_doc.go

```diff
@@ -178,14 +183,32 @@ func convertToInt(propertyName string, untypedValue any) (int64, error) {
 
 // @todo: Implement Encoded Document type
 type encodedDocument struct {
+	mapping *core.DocumentMapping
```

Comment:
suggestion:  This is one change that I am really not a fan of.  I was under the impression that fetcher was something that 'transcended' the planner system, yet here we are making a planner/mapper type a property on a core fetcher type (instead of a param to a couple of planner-centric funcs).

IIRC the only reason `DocumentMapping` is in the `core` package is because it had to be referenced by `fetcher.FetchNextDoc` and `encodedDocument.DecodeToDoc`.  It is very coupled to the planner system and if you really want it as a prop on a fetcher packages type, I would instead suggest that we instead move fetcher out of `db` and into `planner` - I don't like that very much either, but here we are increasing the level of coupling between the two.

Might be worth having a call about this one.

Reply:
Its a hard call, I'd say youre both right and wrong here.

Its needed here as you note because of `fetcher.FetchNextDoc` which returns a `core.Doc` type, which is used by the planner. In reality, the `DocumentMapping` and `core.Doc` could really be used anywhere, as its conceptually only tied to the `CollectionDescription`. 

I was almost going to hoist it into the description just to see what that would look like, as that would mean a `core.Doc` could be used anywhere. Theres nothing specific `core.Doc` has to the planner. The goal of it is to have an effecient array based representation for a document, instead of relying on a `map[string]any` everywhere. Its primarily used in the planner, but as there are other places that document handling needs to be refactored (#935) it could be the case the `core.Doc` is used everywhere.

But thats a bigger conversation that we don't need to get into.
--

Bottom line, the fetcher now needs to be able to run a filter evaluation, it can use the old style of filter evaluation that was used before the mapper existed, but then we would be doing double work to decode the document. Once to a `map[string]any` to run with the old version of `parser.RunFilter`, and once to return for `FetchNextDoc`.

Additionally, when `mapper` was implemented, the `connor` package was migrated to use `map[FilterKey]any` instead of `map[string]any` which is generated in `mapper/targetable.go`.

So it seems its somewhat bound to use the mapper system due to the filter requirements from within the fetcher.

Open to suggestions.
--

One reason to avoid putting `fetcher` in `planner` is that it can be run entirely on its own, as you can see in `db/fetcher_test.go`, and from within the Collection API, which uses the fetcher.

Happy to have a call 📞

Reply:
We spoke about this a while back, and I don't think we'll see eye-to-eye on this any time soon :)  Author view to trump reviewer (ignoring CTO status ;)), and mark this convo as resolved.

I have a plan for the fetcher that I want to try out in the near future anyway, which will pretty much bypass this conflict entirely if it works out.